src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -639,7 +641,14 @@ def load_sub_model(
             f"The component {class_obj} of {pipeline_class} cannot be loaded as it does not seem to have"
             f" any of the loading methods defined in {ALL_IMPORTABLE_CLASSES}."
         )
-
+    if pipeline_class.__name__ == "DiTPipeline":
```

Comment:
Perhaps we use a mapping here so that the code can generalise to other models and pipelines? e.g 

```python
REMAPPING = {
    "DiTPipeline": {
        "Transformer2DModel": DiTTransformer2DModel
    }
}

class_obj = REMAPPING[pipeline_class.__name__][class_obj.__name__]

```

Reply:
Works. But that doesn't likely change the overall approach of the changes being introduced. I was more interested to hear your thoughts on the broader approach being taken here.

Reply:
Plus a check to see if a remapping exists or not.

Reply:
Done.